rust/protocol/src/consts.rs

```diff
@@ -1,10 +1,34 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2022 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
-pub const MAX_FORWARD_JUMPS: usize = 25_000;
-pub const MAX_MESSAGE_KEYS: usize = 2000;
-pub const MAX_RECEIVER_CHAINS: usize = 5;
-pub const ARCHIVED_STATES_MAX_LENGTH: usize = 40;
-pub const MAX_SENDER_KEY_STATES: usize = 5;
+#![warn(missing_docs)]
+
+//! Magic numbers.
+
+/// Various positive integers bounding the maximum size of other data structures.
+pub mod limits {
+    /// The maximum number of encrypted messages that the client chain which decrypts Signal
+    /// messages in a [Double Ratchet] instance can retrieve at once (tracked in
+    /// [crate::proto::storage::session_structure::chain::ChainKey::index] as well as a separate
+    /// `counter`).
+    ///
+    /// [Double Ratchet]: https://signal.org/docs/specifications/doubleratchet/
+    pub const MAX_FORWARD_JUMPS: usize = 25_000;
+    /// The maximum number of per-message keys that can be retained to decrypt messages within
+    /// a specific chain from `message_keys` in [crate::proto::storage::session_structure::Chain].
+    pub const MAX_MESSAGE_KEYS: usize = 2000;
```

Comment:
Similarly, this one is about when you *do* encounter such a jump, how many past keys will the session retain in case you receive some of those missing messages (due to some kind of out-of-order delivery, I guess).

Reply:
I have spent a little time to rewrite this, referring to the documentation for `MAX_FORWARD_JUMPS` I rewrote above.

Reply:
(this is now in `limits.rs`)